rust/src/dcerpc/dcerpc_udp.rs

```diff
@@ -137,8 +148,8 @@ impl DCERPCUDPState {
         if otx.is_none() {
             let ntx = self.create_tx(hdr);
             SCLogDebug!("new tx id {}, last tx_id {}, {} {}", ntx.id, self.tx_id, ntx.seqnum, ntx.activityuuid[0]);
-            self.transactions.push(ntx);
-            otx = self.transactions.last_mut();
+            self.transactions.push_back(ntx);
+            otx = self.transactions.back_mut();
```

Comment:
From what I understand, `Deque` because of it's structure, is good to be used in cases that require pushing and popping from the front. That's what I saw in one of Jason's commit messages as well. However, all I see are too many `push_back` calls in the code and just one `pop_front` call (in `dns`).
So, I'm interested in learning why using `Deque` helps in all the other protos where it exists (including `dcerpc`) in terms of efficiency.

Reply:
VecDeque is good for use when pushing in the back and popping from the front (first in first out)

I think the implementation kind of transforms `remove(0)` into `pop_front` that is why we do not see `pop_front`

This is especially used by `AppLayerParserTransactionsCleanup` which uses `get_tx_iterator` 

Is it clearer ?

Reply:
> I think the implementation kind of transforms `remove(0)` into `pop_front` that is why we do not see `pop_front`
> 

I see. Could you please point to the part of code where this is happening?

> This is especially used by `AppLayerParserTransactionsCleanup` which uses `get_tx_iterator`
> 
> Is it clearer ?

Not really. Tried to see this part and it was too magical FFI, I suppose. But, I do get the concept. Thank you. :)

Reply:
> > I think the implementation kind of transforms `remove(0)` into `pop_front` that is why we do not see `pop_front`
> 
> I see. Could you please point to the part of code where this is happening?
> 

Line 104 : https://github.com/OISF/suricata/blob/86ee4d751e45dda55335520f3a24fea375968870/rust/src/dcerpc/dcerpc_udp.rs#L104
`self.transactions.remove(index);`